lib/queue.js

```diff
@@ -1091,8 +1091,8 @@ Queue.prototype.getNextJob = function() {
           }
         },
         err => {
-          // Swallow error
-          if (err.message !== 'Connection is closed.') {
+          // Swallow error only for connection closed. See GH issue #1304
+          if (err.message === 'Connection is closed.') {
             console.error('BRPOPLPUSH', err);
           }
```

Comment:
```suggestion
          // Ignore 'Connection is closed.' error if closing the queue
          if (!(err.message === 'Connection is closed.' && this.closing)) {
            console.error('BRPOPLPUSH', err);
            throw err;
          }
```

Reply:
@dobesv this is interesting. How about the delay mentioned by @manast? Do we need that, or not? It seems to be part of your suggestion here https://github.com/OptimalBits/bull/issues/1746

Reply:
Yeah, after I proposed this I also thought of a way to sort of throttle attempts if we got an error.  It's hard to say exactly what the best policy is here, we're a bit in the dark.  But for a resilient system we could try to assume that redis is temporarily down or a graceful shutdown was initiated but we haven't been told yet (e.g. closing was not set *yet* but might be very soon).

It seems like maybe some kind of `emit('error')` or something like that to allow the library user to detect this condition might also be useful.